### DISCUSSION TITLE
get_raw_value() to avoid sscanf for float. unsafe_get()

### DISCUSSION THREAD

**Discussion by adrianboston** [taocpp/json#99] [Ideas]
I am seeking a way of accessing the 'raw' value - for lack of a better word - of a value. Specifically of a float or double.

Using the following as an example:

      {
        "address": "A string",
        "age": 12
        "name": "Customer 1",
        "salary": 95000.23
      }

salary.get_double() returns that dang  95000.2300000000002 value.

Given that json is a string at some point can we access the float as a string.

I see 
         double get_double() const
         {
            validate_json_type( json::type::DOUBLE );
            return unsafe_get_double();
         }

Is there just an unsafe_get()

**Comment by d-frey** [comment]
We already have some ideas in that direction, although it will probably not be in the value class, only in the event interface. But I might take the value class into account as well, I just need more time.

**Reply by ColinH** [reply]
To elaborate, while JSON is a text-based format the `value` class stores everything converted to some corresponding "binary" representation. In the case of floating-point numbers they are stored as `double` which means that the original string representation from the JSON input is no longer available.

In the future we might extend our Events interface to selectively use the string representation of numbers instead of only the converted binary representation; once that is done there will be more possibilities including a hypothetical alternative value class that stores numbers as strings.

And btw, though I'm no expert in that particular field I believe that decimal fixed-point representations work better for monetary values than binary floating point since, as you just found out, too, the latter can not represent all decimal fractions precisely.
